product_contract/models/contract.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.exceptions import AccessError
+from odoo.tools.translate import _
+
+
+class ContractContract(models.Model):
+    _inherit = "contract.contract"
+
+    sale_order_count = fields.Integer(compute="_compute_sale_order_count")
+
+    @api.depends("contract_line_ids")
+    def _compute_sale_order_count(self):
+        for rec in self:
+            try:
```

Comment:
Seems that there are no records in self, so maybe there must be a sale order related to the contrat at test...